lib/mixlib/shellout/windows/core_ext.rb

```diff
@@ -530,5 +568,39 @@ def format_creds_from_hash(hash)
       [ logon, passwd, domain ]
     end
 
+    # Retrieves the environment variables for the specified user.
+    #
+    def retrieve_environment_variables(token, env_list)
+      env_pointer = FFI::MemoryPointer.new(:pointer)
+      create_environment_block(env_pointer, token, false)
+      str_ptr = env_pointer.read_pointer
+      offset = 0
+      loop do
+        new_str_pointer = str_ptr.+(offset)
+        break if new_str_pointer.read_string(2) == ENVIRONMENT_BLOCK_ENDS
+
+        environment = new_str_pointer.read_wstring
+        env_list << environment
+        offset = offset + environment.length * 2 + 2
+      end
+      # To free the buffer when we have finished with the environment block
+      destroy_environment_block(str_ptr)
+      env_list
+    end
+
+    # Merge environment variables of specified user and current environment variables.
+    #
+    def merge_env_variables(fetched_env, current_env)
+      env_hash_1 = to_hash(fetched_env)
+      env_hash_2 = to_hash(current_env)
+      merged_env = env_hash_2.merge(env_hash_1)
+      merged_env.map { |k, v| "#{k}=#{v}" }
+    end
+
+    # Convert an array to a hash.
+    #
+    def to_hash(str)
```

Comment:
```suggestion
    def to_hash(str)
```

Since `to_hash` is a common method name in ruby, this could probably use a more descriptive name, something like `environment_list_to_hash`.